app/src/lib/stores/app-store.ts

```diff
@@ -2907,59 +2907,22 @@ export class AppStore extends TypedBaseStore<IAppState> {
       repository
     )
 
-    let stashToPop: IStashEntry | null = null
-    if (enableStashing()) {
-      const hasChanges = changesState.workingDirectory.files.length > 0
-      if (
-        hasChanges &&
-        uncommittedChangesStrategy.kind ===
-          UncommittedChangesStrategyKind.AskForConfirmation
-      ) {
-        this._showPopup({
-          type: PopupType.StashAndSwitchBranch,
-          branchToCheckout: foundBranch,
-          repository,
-        })
-        return repository
-      }
-
-      const { tip } = branchesState
-      const currentBranch = tip.kind === TipState.Valid ? tip.branch : null
-      if (
-        currentBranch !== null &&
-        uncommittedChangesStrategy.kind ===
-          UncommittedChangesStrategyKind.StashOnCurrentBranch
-      ) {
-        await this._createStashAndDropPreviousEntry(
-          repository,
-          currentBranch.name
-        )
-        this.statsStore.recordStashCreatedOnCurrentBranch()
-      } else if (
-        uncommittedChangesStrategy.kind ===
-        UncommittedChangesStrategyKind.MoveToNewBranch
-      ) {
-        const hasDeletedFiles = changesState.workingDirectory.files.some(
-          file => file.status.kind === AppFileStatusKind.Deleted
-        )
-        if (
-          hasDeletedFiles &&
-          uncommittedChangesStrategy.transientStashEntry === null
-        ) {
-          const stashCreated = await gitStore.performFailableOperation(() => {
-            return createDesktopStashEntry(repository, foundBranch.name)
-          })
-
-          if (stashCreated) {
-            stashToPop = await getLastDesktopStashEntryForBranch(
-              repository,
-              foundBranch.name
-            )
-          }
-        }
-      }
+    const hasChanges = changesState.workingDirectory.files.length > 0
+    if (hasChanges && uncommittedChangesStrategy.kind === askToStash.kind) {
+      this._showPopup({
+        type: PopupType.StashAndSwitchBranch,
+        branchToCheckout: foundBranch,
+        repository,
+      })
+      return repository
     }
 
+    let stashToPop: IStashEntry | null = await this._stashBasedOnUncommittedChangesStrategy(
```

Comment:
With this extraction, it would be awesome to make this a `const` and minimize mutation if not completely necessary. Which I think we should be able to do with:

```
diff --git a/app/src/lib/stores/app-store.ts b/app/src/lib/stores/app-store.ts
index 0d7d02af4..152dbc75a 100644
--- a/app/src/lib/stores/app-store.ts
+++ b/app/src/lib/stores/app-store.ts
@@ -2958,9 +2958,10 @@ export class AppStore extends TypedBaseStore<IAppState> {
       // call this method again once the working directory has been cleared.
       this.statsStore.recordChangesTakenToNewBranch()
 
-      stashToPop = stashToPop || uncommittedChangesStrategy.transientStashEntry
-      if (stashToPop !== null) {
-        const stashSha = stashToPop.stashSha
+      const stashEntry =
+        stashToPop || uncommittedChangesStrategy.transientStashEntry
+      if (stashEntry !== null) {
+        const stashSha = stashEntry.stashSha
         await gitStore.performFailableOperation(() => {
           return popStashEntry(repository, stashSha)
         })
```

This'll make it easier to scan the code and know with certainty what `stashToPop` is (since it's value is assigned in only one place) and we can ensure that we don't add logic in the future that changes the value in unexpected ways.

Reply:
yeah, i agree this would be a nice improvement too! also agree its non-blocking.

fwiw this line was only moved down from earlier in the function by this PR (see https://github.com/desktop/desktop/pull/8087/files#diff-d6b76b602a5c48740d6d10deda6f71ebL2910).